### PR TITLE
Introduction of Promises

### PR DESCRIPTION
**Pull request by riskawarrior**
* Promises are added backward compatible
* Old beta node version is dropped, latest stable is added
* Docs and tests are extended

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,13 +63,24 @@ function (value) {
...
 Ask for a name:
+
 ```js
 promptly.prompt('Name: ', function (err, value) {
     // err is always null in this case, because no validators are set
     console.log(value);
 });
 ```
 
+Using Promise:
```
Comment (satazor) [inline README.md]: Could you please remove all the "Using promise:" examples and just mention it once? perhaps at the top saying that all functions work with promises?

File: README.md [hunk 2]
```diff
@@ -62,13 +63,24 @@ function (value) {
...
+
+```js
+promptly.prompt('Name: ')
+    .then(function (value) {
```
Comment (satazor) [inline README.md]: I would remove indentation here an keep it 0 level (for the top example).

File: test/test.js [hunk 3]
```diff
@@ -189,6 +189,54 @@ describe('prompt()', function () {
 
         sendLine('yeaa');
     });
+
+    it('should prompt the user (using promise)', function (next) {
```
Comment (satazor) [inline test/test.js]: Same thing for the tests.. I would only add one test per function, that actually test the promise usage. Could you change that?

**Comment by satazor** [comment]
@riskawarrior Thanks for the PR, I've reviewed it.

File: test/test.js [hunk 4]
```diff
@@ -189,6 +189,54 @@ describe('prompt()', function () {
...
+        promptly.prompt('something: ')
+            .then(function (value) {
```
Comment (satazor) [inline test/test.js]: Can you keep it 0 level indentation? Same for all promise usage.

**Comment by riskawarrior** [comment]
@satazor Sure, I made the alterations you suggested. Please review it again, thanks!

**Comment by satazor** [comment]
Looks good now! Thanks

**Pull request merged**